R/utilities-units.R

```diff
@@ -136,9 +196,47 @@ getDimensionForUnit <- function(unit) {
 getUnitsForDimension <- function(dimension) {
   validateIsString(dimension)
   dimensionTask <- getDimensionTask()
-  rClr::clrCall(dimensionTask, "AllAvailableUnitNamesForDimension", dimension)
+  rClr::clrCall(dimensionTask, "AllAvailableUnitNamesFor", dimension)
+}
+
+#' Get conversion factor from one unit to another
+#'
+#' @param fromUnit String name of the unit to convert from
+#' @param toUnit String name of the unit to convert to
+#' @param dimension Dimension of the units (conversion between mass and molar possible when
+#' \code{MW} is provided)
+#' @param MW Optional molecule weight in g/mol to use when converting, for example, from molar to mass amount or concentration

```

Comment:
I think MW should be passed as g/mol, this is the only unit a user expects.
@msevestre

Reply:
>MW Optional molecule weight in g/mol

yes?

Reply:
Yes? :D

Reply:
Ok I see your point. The problem is that we always pass parameters in core unit
height should be dm etc.. 
Introducing a change like this would not be logical, specifically because the weight generally comes from simulation$molWeightFor which returns in core unit.
I am still not 100% sure I understand why we need such a method. If this is just for the MW, then we could add another flag to toUnit (mw.in.base.unit = true) by default or sthg like this

Reply:
No its not for the MW, its for conversion of values that are not in base units - and most probably these would be the values that do not come from a simulation (that can be converted with toUnit). It is for data that are defined in other units - loaded from excel, generated by hand, or whatsoever.. And in this case MW most definitely will be in g/mol (even in the data exporter we always assume MW is in g/mol).

I see your point that this will be inconsistent with the rest of methods. But I also do not see the use case for this function with MW being in the base unit.

Reply:
LEt's discuss WEdnesday. I do not understand why a user would not convert an array using toUnit instead of dealing with some weird factors